client/js/store/modules/auth.js

```diff
@@ -24,64 +52,22 @@ export default {
     },
   },
   actions: {
-    requestLogin(context, data) {
-      return new Promise((resolve, reject) => {
-        io.socket.post(
-          '/user/login',
-          {
-            username: data.username,
-            password: data.password,
-          },
-          function handleResponse(resData, jwres) {
-            if (jwres.statusCode === 200) {
-              context.commit('authSuccess', data.username);
-              return resolve();
-            }
-            context.commit('clearAuth');
-            return reject(jwres.body.message);
-          }
-        );
-      });
+    async requestLogin(context, { username, password }) {
+      await handleLogin(context, username, password);
     },
 
-    requestSignup(context, data) {
-      return new Promise((resolve, reject) => {
-        io.socket.put(
-          '/user/signup',
-          {
-            username: data.username,
-            password: data.password,
-          },
-          function handleResponse(resData, jwres) {
-            if (jwres.statusCode === 200) {
-              context.commit('authSuccess', data.username);
-              return resolve();
-            }
-            let message;
-            if (Object.prototype.hasOwnProperty.call(resData, 'message')) {
-              message = resData.message;
-            } else if (typeof resData === 'string') {
-              message = resData;
-            } else {
-              message = new Error('Unknown error signing up');
-            }
-            context.commit('clearAuth');
-            return reject(message);
-          }
-        );
-      });
+    async requestSignup(context, { username, password }) {
+      await handleLogin(context, username, password, true);
     },
 
-    requestLogout(context) {
-      return new Promise((resolve, reject) => {
-        io.socket.get('/user/logout', {}, function handleResponse(resData, jwres) {
-          if (jwres.statusCode === 200) {
-            context.commit('clearAuth');
-            return resolve();
-          }
-          return reject(new Error('Error logging out :('));
+    async requestLogout(context) {
+      try {
+        await fetch('/user/logout', {
+          credentials: 'include',
```

Comment:
I think the same applies for log out
```suggestion
        return await fetch('/user/logout', {
          credentials: 'include',
```